js/index.js

```diff
@@ -0,0 +1,128 @@
+/* jshint browser: true, browserify: true */
+
+(function () {
+  'use strict';
+
+  var avsc = require('avsc'),
+      buffer = require('buffer'),
+      $ = require('jquery');
+
+  require('jquery-ui');
+  window.avsc = avsc;
+
+  $( function() {
+    
+    resize();
+    /* Change the color if this is the active textarea*/
+    $('.textbox').click( function() { 
+      $(this).addClass('editing');
+    }).blur(function() {
+      $(this).removeClass('editing');
+     });
+
+    /* Validate schema after each new character. */
+    $('#schema').keyup(function(event) {
+        validateSchema();
+    });
+    function validateSchema() {
+      var elem = $('#schema');
+      var error_elem = $('#schema-error');
+      try {
+        var schema = readInput('#schema');
+        var p = avsc.parse(schema);
+        error_elem.removeClass('error');
+        error_elem.addClass('valid');
+        error_elem.text('Valid Schema!');
+      } catch (err) {
+        showError(error_elem, err);
+      }
+    }
+ 
+    $('#encode').click(function() {
+      var schema = readInput('#schema');
```

Comment:
What do you think of saving the parsed schema into a variable, and using its current value when a decode/encode interaction comes in? (Only parsing it when the `#schema` element changes.)